### PR TITLE
Convert confirm-tx.utils.js to typescript

### PR DESCRIPTION
**Pull request by brad-decker**
## Explanation
Straightforward conversion to TypeScript, added the types definition for the currency-formatter module and removed an option from its usage that appeared to be incorrect even for the version of currency formatter we use now. 

## Manual Testing Steps
1. review changes to ensure the types make sense
2. confirm tests pass

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

**Comment by brad-decker** [comment]
@jpuri -- might be relative to your current interests, I see you are refactoring the confirmations

File: ui/helpers/utils/confirm-tx.util.ts [hunk 1]
```diff
@@ -74,18 +86,28 @@ export function convertTokenToFiat({
...
  * Rounds the given decimal string to 4 significant digits.
  *
- * @param {string} decimalString - The base-ten number to round.
- * @returns {string} The rounded number, or the original number if no
+ * @param decimalString - The base-ten number to round.
+ * @returns The rounded number, or the original number if no
  * rounding was necessary.
  */
-export function roundExponential(decimalString) {
+export function roundExponential(decimalString: string) {
```
Comment (DDDDDanica) [inline ui/helpers/utils/confirm-tx.util.ts:110]: Can we add a return type here?

File: ui/helpers/utils/confirm-tx.util.ts [hunk 2]
```diff
@@ -95,8 +117,10 @@ export function roundExponential(decimalString) {
     : decimalString;
 }
 
-export function areDappSuggestedAndTxParamGasFeesTheSame(txData = {}) {
-  const { txParams, dappSuggestedGasFees } = txData;
+export function areDappSuggestedAndTxParamGasFeesTheSame(
+  txData: TransactionMeta,
+) {
```
Comment (DDDDDanica) [inline ui/helpers/utils/confirm-tx.util.ts:122]: would be great to add boolean as return type here ~

File: ui/helpers/utils/confirm-tx.util.ts [hunk 3]
```diff
@@ -45,23 +52,28 @@ export function getTransactionFee({
   return fee.round(numberOfDecimals).toString();
 }
 
-export function formatCurrency(value, currencyCode) {
+export function formatCurrency(value: string, currencyCode: string) {
```
Comment (DDDDDanica) [inline ui/helpers/utils/confirm-tx.util.ts:55]: Can you add a return type here? :)

File: ui/helpers/utils/confirm-tx.util.ts (hunk 1 above)
Comment (brad-decker) [reply ui/helpers/utils/confirm-tx.util.ts]: Done in latest

File: ui/helpers/utils/confirm-tx.util.ts (hunk 2 above)
Comment (brad-decker) [reply ui/helpers/utils/confirm-tx.util.ts]: Done in latest

File: ui/helpers/utils/confirm-tx.util.ts (hunk 3 above)
Comment (brad-decker) [reply ui/helpers/utils/confirm-tx.util.ts]: Done in latest

**Review by DDDDDanica — APPROVED** [review]

**Review by digiwand — APPROVED** [review]

**Pull request merged**